### PR TITLE
Test for building libsovl (fix #62)

### LINKED ISSUE #62: try build libsolv package with RPG
follow the process of RPG (all the phases: see /tmp/rpg-*) and watch warnings -> add issues/fix it
the testing project is libsolv project folder

### PR DESCRIPTION
**Pull request by cibo94**
This run of first try - there was no problem

### CODE REVIEW

**Comment by jsilhan** [comment]
this looks like a dream. Can you add into `spec.build` these cmake flags and see if it found all dependencies, please:
```bash
-DCMAKE_BUILD_TYPE=RelWithDebInfo \\\
            -DENABLE_PERL=1 \\\
            -DENABLE_PYTHON=1 \\\
            -DENABLE_RUBY=1 \\\
            -DUSE_VENDORDIRS=1 \\\
            -DFEDORA=1 \\\
            -DENABLE_LZMA_COMPRESSION=1 \\\
```

(without any switch it probably has no dependencies. we build it for Fedora with these cmake params).

**Comment by cibo94** [comment]
I had problems with the version you have provided so I just use self.base.load_project_from_url("https://github.com/openSUSE/libsolv/") and it works just fine 'till I hit the ```make install DESTDIR=/builddir/build/BUILDROOT/LibSolv-0.6.11-1.fc22.x86_64``` zone in mock build where I get this error:
```
[ 94%] Building C object bindings/perl/CMakeFiles/bindings_perl.dir/solv_perl.o
/builddir/build/BUILD/LibSolv-0.6.11/bindings/perl/solv_perl.c:732:20: fatal error: EXTERN.h: No such file or directory
compilation terminated.
bindings/perl/CMakeFiles/bindings_perl.dir/build.make:60: recipe for target 'bindings/perl/CMakeFiles/bindings_perl.dir/solv_perl.o' failed
CMakeFiles/Makefile2:784: recipe for target 'bindings/perl/CMakeFiles/bindings_perl.dir/all' failed
```

I think that it will be nice to catch build errors in C plugin and add missing files to required files then rebuild again.

**Comment by cibo94** [comment]
BTW ```EXTERN.h``` is header that provides embedding perl.

File: tests/unit/test_functional.py [hunk 1]
```diff
@@ -18,25 +18,6 @@ def test_c_project(self):
         base.spec.build = "make"
         base.run_extracted_source_analysis()
         base.run_patched_source_analysis()
-        expected_required_files = {
```
Comment (jsilhan) [inline tests/unit/test_functional.py]: remove packages that are not the same for i686  arch and check that this files are contained in the result after analysis, please.

**Comment by jsilhan** [comment]
Merged, thanks

**Pull request closed without merging**